x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -57,16 +58,17 @@ export class EnterpriseSearchPlugin implements Plugin {
       appRoute: ENTERPRISE_SEARCH_PLUGIN.URL,
       category: DEFAULT_APP_CATEGORIES.enterpriseSearch,
       mount: async (params: AppMountParameters) => {
-        const [coreStart] = await core.getStartServices();
-        const { chrome } = coreStart;
-        chrome.docTitle.change(ENTERPRISE_SEARCH_PLUGIN.NAME);
+        const kibanaDeps = await this.getKibanaDeps(core, params);
+        const pluginData = this.getPluginData();
 
-        await this.getInitialData(coreStart.http);
+        const { chrome, http } = kibanaDeps.core;
+        chrome.docTitle.change(ENTERPRISE_SEARCH_PLUGIN.NAME);
+        await this.getInitialData(http);
 
         const { renderApp } = await import('./applications');
         const { EnterpriseSearch } = await import('./applications/enterprise_search');
 
-        return renderApp(EnterpriseSearch, params, coreStart, plugins, this.config, this.data);
+        return renderApp(EnterpriseSearch, kibanaDeps, pluginData);
```

Comment:
Hopefully this is now easier to read/conceptualize than the old row of args that got away from me, haha.